tests/compiler/TestHipccFp16Include.bash

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+# Test the case where the hip/hip_fp16.h header is included before hip/hip_runtime.h 
+set -eu
+
+SRC_DIR=@CMAKE_CURRENT_SOURCE_DIR@
+OUT_DIR=@CMAKE_CURRENT_BINARY_DIR@/@TEST_NAME@.d
+HIPCC=@CMAKE_BINARY_DIR@/bin/hipcc
+
+export HIPCC_VERBOSE=7
+mkdir -p ${OUT_DIR}
+
+${HIPCC} ${SRC_DIR}/inputs/testfp16include.cpp
```

Comment:
Do we have other test that also produces `a.out` in the same directory? If so, this test could clash with it when the tests are run parallel. Easily avoided by redirecting the output to /dev/null since this test is not interested in it.